### PR TITLE
feat(ts): allow passing srs via env functions

### PR DESCRIPTION
**Pull request by ludamad**
# Description

This will hopefully allow streamlining situations where the c-bind API is being implemented for typescript. It will allow typescript to simply define how to grab SRS, and then will not need to pass it past that.

Some minor updates to the ts/ folder as well.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/barretenberg/env/crs.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include "barretenberg/ecc/curves/bn254/scalar_multiplication/c_bind.hpp"
+
+const int NUM_POINTS_IN_TRANSCRIPT = 5040001;
+
+
+extern "C" {
+/**
+ * @brief In WASM, loads the verifier reference string.
+ * Used in native code to quickly create an in-memory reference string.
+ * @param circuit_size The size of the circuit.
```
Comment (dbanks12) [inline cpp/src/barretenberg/env/crs.cpp]: This param isnt there

Comment (dbanks12) [reply cpp/src/barretenberg/env/crs.cpp]: same a few times below

File: cpp/src/barretenberg/srs/reference_string/env_reference_string.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Create reference strings given an environment that implements env/crs.hpp.
+ * Usable in both native and WASM code, but particularly useful for accessing
+ * reference strings from a WASM context. 
+ * In a native context, the implementations assume a hard-coded string path.
+ * For that reason, ideally this is only used in c-bind functions to maintain flexibility.
```
Comment (dbanks12) [inline cpp/src/barretenberg/srs/reference_string/env_reference_string.hpp:6]: Solid comment, thanks

File: ts/src/crs/index.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
+import { readFile } from "fs/promises";
```
Comment (dbanks12) [inline ts/src/crs/index.ts:1]: I think I see this file in a3-circuit/ts too. Does it belong in both?

**Review by dbanks12 — CHANGES_REQUESTED** [review]
Looks good. Just really one comment that needs fixing.

File: ts/src/crs/index.ts (hunk 3 above)
Comment (ludamad) [reply ts/src/crs/index.ts:1]: It belongs here. I will remove the redundancy soon.

**Review by dbanks12 — APPROVED** [review]
LGTM

**Pull request merged**